### DISCUSSION TITLE
Required status checks, reusable workflows and skipped jobs

### DISCUSSION THREAD

**Discussion by markgoddard** [community/community#72708] [Actions]
### Select Topic Area

Product Feedback

### Body

I have a [PR workflow](https://github.com/stackhpc/stackhpc-kayobe-config/blob/dae64ddbc692d87a4b156e042b66eb65c104d7f6/.github/workflows/stackhpc-pull-request.yml) that calls several other reusable workflows. There is a dependency from the `all-in-one-*` jobs on the `build-kayobe-image` job. Each job is listed as a required status check.

Since these jobs take a long time, I'm trying to add path filtering to run certain jobs when particular files changed.

At first, I tried the [workflow-level path filtering](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore). This is a nice feature but does not work with my reusable workflows because the name of the check differs depending on whether the job is skipped or not.

e.g. When the `all-in-one-centos-ovs` job in the outer workflow runs, it registers a check as `aio (CentOS OVS) / All in one`. However if the job is skipped, it registers as `aio (CentOS OVS)` and I still see the `aio (CentOS OVS) / All in one` job as `Expected — Waiting for status to be reported`.

Next, I tried using the [dorny/paths-filter](https://github.com/dorny/paths-filter) action to conditionally skip jobs. This suffers from the same problem with job naming, but I was able to work around it by [passing a skip condition into the reusable job](https://github.com/stackhpc/stackhpc-kayobe-config/pull/708/commits/6cf2d27819dacac8cbf92df58a9a8454577656b5).

I also had to add an [`always()`](https://github.com/stackhpc/stackhpc-kayobe-config/pull/708/commits/b075159e47f2b257103569fe6e43d72d4d4cd0a6) condition to the dependent jobs to avoid them being skipped if the parent job is skipped. I'm unsure if this is a good solution.

I believe this is now doing what I want, but it doesn't seem like an ideal solution.

**Comment by markgoddard** [comment]
Follow up: `always()` didn't work in the case where the parent job failed, so I switched to `! failure()`, and this seems to be working. [Here](https://github.com/stackhpc/stackhpc-kayobe-config/pull/726/commits/57021fec0c559c04e2d9f20b93a2f0e4e179d826) is the result.

**Comment by dja557** [comment]
I have the same problem. I ended up creating my own status checks using the GH CLI based on success or not. This works but again, not an ideal solution at all.

**Comment by Fresa** [comment]
Agree, skipped jobs are not being reported with the reusable workflow job's name included. I guess it's never evaluated when the calling job's condition is not fulfilled. 
![image](https://github.com/user-attachments/assets/a092fe2f-1882-47e2-9baa-14363872aa70)

These two are "the same check", the first one get's reported when the job is skipped, the second one when it runs.

**Comment by LukeCarrier** [comment]
GitHub, I loathe and despise you. GitHub Actions is such a foresaken mess.
